### PR TITLE
[ipython] implement `canvas.Update()` in notebooks

### LINKED ISSUE #13631: LiveVisualize does not update canvas on Jupyter notebook
### Explain what you would like to see improved and how.

The LiveVisualize feature in ROOT does not update the canvas when used in Jupyter Notebooks, resulting in the canvas not being continuously refreshed with new plots as expected.

### ROOT version

v6.29/01

### Installation method

Compiled from source

### Operating system

Fedora 38

### Additional context

**Repro:**

```
if __name__ == "__main__":

    connection = create_connection()
    num_entries = 1000000
    d = RDataFrame(num_entries, daskclient=connection, npartitions=5)
    dd = d.Define("x", "rdfentry_").Define("y", "x*x")
    
    h = dd.Histo1D(("hist", "hist", 100, 0, 20), "x") 

    LiveVisualize([h])
    
    c = ROOT.TCanvas()
    c.Draw()
    h.Draw()
    c.Update()
```

**Expected Result:**

The canvas should continuously update with new histogram values, reflecting partial results being merged in real-time.

**Actual Result:**

The canvas does not update during the execution, and the final result is only displayed after the computation graph completes.

### PR DESCRIPTION
**Pull request by linev**
1. Add flag in TCanvas/RCanvas which set when `Update()` method is called.
2. Use `can.IsUpdated()` flag in iPython to check if canvas was updated
3. Use iPython display handle to update drawing
4. Works with JSROOT and png-based drawings
5. Use latest released JSROOT version 7.4.3
6. Use TWebCanvas JSON for TCanvas drawings to handle all special cases like style, list of colors, stats box and so on

Resolves #13631

### CODE REVIEW

File: graf2d/gpad/inc/TCanvas.h [hunk 1]
```diff
@@ -61,6 +61,7 @@ friend class TInterpreter;
    Bool_t        fRetained;        ///< Retain structure flag
    Bool_t        fUseGL;           ///<! True when rendering is with GL
    Bool_t        fDrawn;           ///<! Set to True when the Draw method is called
+   Bool_t        fUpdated;         ///<! Set to True when Update method was called
```
Comment (dpiparo) [inline graf2d/gpad/inc/TCanvas.h:64]: @couet ?

**Review by dpiparo — COMMENTED** [review]
Thanks for this code, @linev ! What perhaps is missing are some tests. Is it possible to test the new functionality?

**Comment by linev** [comment]
> @dpiparo  What perhaps is missing are some tests. Is it possible to test the new functionality?

No idea how it can be tested. It changes existing drawing, but how it can be verified?

**Review by dpiparo — APPROVED** [review]
Thanks again Serguei for these changes. An general observation, which is not at all specific to this PR, could be that it would be extremely useful to be in position to "test the graphics" for example comparing some features of the images obtained from plots in some friendly format...

**Comment by linev** [comment]
> "test the graphics"

For web-based graphics it is done here: https://github.com/linev/jsroot-test

Same in browser: https://root.cern/js/dev/examples.htm?more#run

It produces ~650 different images and compares every single line or/and pixel.
Also now it includes interactive features testing - also inside `node.js`

**Pull request merged**